semantic-conventions/semconv.schema.json

```diff
@@ -0,0 +1,302 @@
+{
```

Comment:
Have you verified that all YAML files that we have in the spec repo pass the validation?

Reply:
No, I don't know a tool that could do this validation. I'd need to open all files in VS code and look at the "problems" view I guess. The main purpose of the file is as an editing aid.

Reply:
The semantic conventions are currently missing the `type:` property which should probably be fixed because at the moment the tool will then default to `span`, even for resources.

You can see warning messages if you look at the semantic convention checkers build output, e.g. here: https://github.com/open-telemetry/opentelemetry-specification/runs/3074608868

```
Using default SPAN type for semantic convention 'aws.ecs' @ line 5
Using default SPAN type for semantic convention 'aws.eks' @ line 5
Using default SPAN type for semantic convention 'aws.log' @ line 5
Using default SPAN type for semantic convention 'container' @ line 5
Using default SPAN type for semantic convention 'deployment' @ line 5
```

etc.

Reply:
I checked all files (using VS code) and found no further problems in them. However, I then noticed a misspelling in the schema that caused something not to be checked, which I fixed. I also made the type optional to reflect reality (maybe we even need to remove `type` from the tool again if we want to share metric and span conventions). 583e29d